CodePush.ios.js

```diff
@@ -126,72 +126,106 @@ function checkForUpdate() {
  * releases, and displaying a standard confirmation UI to the end-user
  * when an update is available.
  */
-function sync(options = {}) {  
+function sync(options = {}, syncStatusChangeCallback, downloadProgressCallback) {  
   var syncOptions = {
-    descriptionPrefix: " Description: ",
-    appendReleaseDescription: false,
     
     ignoreFailedUpdates: true,
-    
-    mandatoryContinueButtonLabel: "Continue",
-    mandatoryUpdateMessage: "An update is available that must be installed.",
-    
-    optionalIgnoreButtonLabel: "Ignore",
-    optionalInstallButtonLabel: "Install",
-    optionalUpdateMessage: "An update is available. Would you like to install it?",
-    
+    installMode: CodePush.InstallMode.ON_NEXT_RESTART,
     rollbackTimeout: 0,
-    
-    updateTitle: "Update available",
+    updateDialog: null,
     
     ...options 
   };
-
+  
+  syncStatusChangeCallback = typeof syncStatusChangeCallback == "function"
+    ? syncStatusChangeCallback
+    : function(syncStatus) {
+        switch(syncStatus) {
+          case CodePush.SyncStatus.CHECKING_FOR_UPDATE:
+            console.log("[CodePush] Checking for update.");
+            break;
+          case CodePush.SyncStatus.DOWNLOADING_PACKAGE:
+            console.log("[CodePush] Downloading package.");
+            break;
+          case CodePush.SyncStatus.AWAITING_USER_ACTION:
+            console.log("[CodePush] Awaiting user action.");
+            break;
+          case CodePush.SyncStatus.INSTALLING_UPDATE:
+            console.log("[CodePush] Installing update.");
+            break;
+          case CodePush.SyncStatus.IDLE:
```

Comment:
Also, I feel like it's hard to keep track of `SyncStatus` vs `SyncResult` names. Would it make sense to combine the two enums into one, and send all of them in the progress callback, but only the final ones in the result? That way, we avoid situations like this one where we have to come up with two names for the same state. (`IDLE` vs `UPDATE_INSTALLED`)